cypress/integration/tests/deals/terms.e2e.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Given,
+  Then,
+  When,
+} from "@badeball/cypress-cucumber-preprocessor/methods";
+import { EditingCard } from "../components/editingCard.e2e";
+
+const UPDATED = "(updated)";
+const MAX_NUMBER_OF_CLAUSES = 10;
+
+class Terms {
+  static addClauseButton() {
+    return cy.get("[data-test='addClauseButton']");
+  }
+  static addClause() {
+    this.addClauseButton().scrollIntoView().click();
+  }
+  static getClauses() {
+    return cy.get("[data-test='termsClause']");
+  }
+}
```

Comment:
FYI: "Page Object" test pattern

Here, I used the Page Object pattern.
But as you can read in the above title [0], it has some drawbacks.
I introduced it here for the short term benefit of code reduction.
In the long term I'm eyeing to transition to the so called "Screenplay pattern" [1].
(Also because the Screenplay pattern is a more "SOLID" Page Object pattern, I'd like to learn the drawbacks of Page Objects myself)

[0]: 
Stop using Page Objects and Start using App Actions
https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/

[1]: 
- https://www.infoq.com/articles/Beyond-Page-Objects-Test-Automation-Serenity-Screenplay/
- https://cucumber.io/blog/bdd/understanding-screenplay-(part-1)/
- https://www.notion.so/curvelabs/Screenplay-Pattern-a717ca7dbd1248989e5c4c2584c8b50b

@dkent600 @paweljanicki @giladt @IonelLupu

Reply:
Looks interesting and cleaner from my point of view. Nothing to say much. I will need to use it to understand it better